keanu-project/src/test/java/io/improbable/keanu/network/ProtobufTest.java

```diff
@@ -0,0 +1,46 @@
+package io.improbable.keanu.network;
+
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class ProtobufTest {
+
+    @Test
+    public void youCanSaveAndLoadANetworkWithValues() throws IOException {
```

Comment:
This is a good basic integration test but we could do a lot more here. Creating nested classes that represent problematic vertex annotations would be a good one. Also, making sure that the BayesianNetwork object can be saved using it's helper methods.

Reply:
Our real vertices provide some decent test cases already it turns out!  Would be good to chat briefly to see exactly what test cases you're looking for - stuff like negative test cases (like hand crafted negative Protobufs?)